contracts/minter-suite/Minters/MinterDAExpRefund/MinterDAExpRefundV0.sol

```diff
@@ -0,0 +1,997 @@
+            // refund mechanism is provided for refunds, unrelated to msg.value
+            _splitETHRevenues(_projectId, currentPriceInWei);
+        } else {
+            // increment the number of refundable invocations that will be
+            // claimable by the artist and admin once auction is validated.
+            // do not split revenue here since will be claimed at a later time.
+            _projectConfig.numRefundableInvocations++;
+        }
+
+        return tokenId;
+    }
+
+    /**
+     * @notice Refunds the sender's payment above current amount due for
+     * project `_projectId`. The current amount due is the the price paid for
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to msg.sender.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefund(uint256 _projectId) external {
+        claimRefundTo(payable(msg.sender), _projectId);
+    }
+
+    /**
+     * @notice Refunds the sender's payment above current amount due for
+     * project `_projectId`. The current amount due is the the price paid for
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to address `_to`.
+     * @param _to Address to send refund to.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefundTo(address payable _to, uint256 _projectId)
+        public
+        nonReentrant
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Receipt storage receipt = receipts[msg.sender][_projectId];
+        // CHECKS
+        // input validation
+        require(_to != address(0), "No refund to the zero address");
+        // require that a user has purchased at least one token on this project
+        require(receipt.numPurchased > 0, "No purchases made by this address");
+        // get the latestPurchasePrice, which returns the sellout price if the
+        // auction sold out before reaching base price, or returns the base
+        // price if auction has reached base price and artist has withdrawn
+        // revenues.
+        // @dev if user is elligible for a refund, they have purchased a token,
+        // therefore we are guaranteed to not have a populated
```

Comment:
Nit: would you mind adding an extra note here please? I wasn't sure why (for an ongoing auction) wouldn't `latestPurchasePrice` be populated if the user bought a token.

Reply:
good callout - I've updated this comment to appropriately say `therefore we are guaranteed to have a populated...` - the `not` was incorrect, but code logic is correct 🙏  (also updated by misspelling of eligible 😓)